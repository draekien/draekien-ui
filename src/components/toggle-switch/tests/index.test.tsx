/** @jsx jsx */
import { jsx } from 'theme-ui';
import { fireEvent, getByText, render } from '@testing-library/react';
import { ToggleSwitch } from '..';

describe('ToggleSwitch component', () => {
  test('should match snapshot', () => {
    const { container } = render(<ToggleSwitch id="toggle" />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when position right', () => {
    const { container } = render(
      <ToggleSwitch id="toggle" togglePosition="right" />
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot and render children', () => {
    const { container } = render(
      <ToggleSwitch id="toggle" togglePosition="right">
        Test
      </ToggleSwitch>
    );

    expect(container).toMatchSnapshot();
  });

  test('should call callback fn when toggled', (done) => {
    const fn = jest.fn(() => done());

    const component = render(
      <ToggleSwitch id="toggle" togglePosition="right" onToggle={fn}>
        Test
      </ToggleSwitch>
    );

    fireEvent(
      getByText(component.container, 'Test'),
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );
  });
});
