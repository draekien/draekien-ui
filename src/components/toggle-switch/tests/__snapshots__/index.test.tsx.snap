// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch component should match snapshot 1`] = `
<div>
  <div
    class="css-1adqeri-ToggleSwitch"
  >
    <div
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 10px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 20px; width: 40px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 10px; cursor: pointer; transition: background 0.25s;"
      />
      <div
        class="react-switch-handle"
        style="height: 18px; width: 18px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        id="toggle"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
    <label
      class="css-1adg6km-ToggleSwitch"
      for="toggle"
    />
  </div>
</div>
`;

exports[`ToggleSwitch component should match snapshot and render children 1`] = `
<div>
  <div
    class="css-1adqeri-ToggleSwitch"
  >
    <label
      class="css-1adg6km-ToggleSwitch"
      for="toggle"
    >
      Test
    </label>
    <div
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 10px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 20px; width: 40px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 10px; cursor: pointer; transition: background 0.25s;"
      />
      <div
        class="react-switch-handle"
        style="height: 18px; width: 18px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        id="toggle"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </div>
</div>
`;

exports[`ToggleSwitch component should match snapshot when position right 1`] = `
<div>
  <div
    class="css-1adqeri-ToggleSwitch"
  >
    <label
      class="css-1adg6km-ToggleSwitch"
      for="toggle"
    />
    <div
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 10px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 20px; width: 40px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 10px; cursor: pointer; transition: background 0.25s;"
      />
      <div
        class="react-switch-handle"
        style="height: 18px; width: 18px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        id="toggle"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </div>
</div>
`;
